### PR TITLE
Register Page (FAD-5683)

### PR DESCRIPTION
**Pull request by jgzamora**

### CODE REVIEW

File: src/config/routes.js [hunk 1]
```diff
@@ -73,6 +74,11 @@ export default [
     public: true,
     component: AuthPage
   },
+  {
+    path: '/#/register',
```
Comment (jonambas) [inline src/config/routes.js]: Why `#/register`?

File: src/pages/register/RegisterUserForm.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          component={TextFieldWrapper}
+          label='Password'
+          helpText='You password must have at least 8 characters and include 1 letter, 1 number, and 1 non-alphanumeric character.'
+          validate={[required, minLength(8)]}
+          warn={[hasLetter, hasNumber, specialCharacter, endsWithWhitespace]}
+          disabled={submitting}
+          type='password'
+          autoComplete='new-password'
+        />
+        <p><small>By joining, you agree to SparPost's <UnstyledLink>Terms of Use</UnstyledLink></small></p>
```
Comment (jonambas) [inline src/pages/register/RegisterUserForm.js]: to `https://www.sparkpost.com/policies/tou/`

Could maybe put this link in config

File: src/pages/register/RegisterUserForm.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        <Button primary submit disabled={submitting}>{ submitting ? 'Joining' : 'Join SparkPost' }</Button>
```
Comment (jonambas) [inline src/pages/register/RegisterUserForm.js]: I'd maybe use "Register/Registering" to match the route & panel title

File: src/pages/register/RegisterPage.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    return (
+      <div>
+        <div className={styles.LogoWrapper}>
+          <a href='https://www.sparkpost.com' title='SparkPost'>
+            <SparkPost.Logo />
+          </a>
+        </div>
+
+        <Panel sectioned accent title='Register'>
+          { this.renderRegisterPanel() }
```
Comment (jonambas) [inline src/pages/register/RegisterPage.js]: Could we add the same copy somewhere
> You have been invited by {email} to join ~the world's most powerful email platform~ SparkPost.

File: src/pages/register/RegisterUserForm.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+          validate={required}
+          disabled={submitting}
+        />
+        <Field
+          name='password'
+          component={TextFieldWrapper}
+          label='Password'
+          helpText='You password must have at least 8 characters and include 1 letter, 1 number, and 1 non-alphanumeric character.'
+          validate={[required, minLength(8)]}
+          warn={[hasLetter, hasNumber, specialCharacter, endsWithWhitespace]}
```
Comment (jonambas) [inline src/pages/register/RegisterUserForm.js]: Do we need `endsWithWhitespace`? Could we trim or is that bad practice

**Comment by jonambas** [comment]
Can we change this too 
> message: 'Invitation is on its way! Once your user signs up, you\'ll see them ~below~ on this page'

File: src/pages/register/RegisterUserForm.js (hunk 5 above)
Comment (jgzamora) [reply src/pages/register/RegisterUserForm.js]: The problem is when they use a whitespace at the end as the special character, which technically works, but the input would then trim it automatically (I think?) and the password would no longer be valid.

File: src/pages/register/RegisterUserForm.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    } = this.props;
+
+    return (
+      <form onSubmit={handleSubmit}>
+        <Field
+          name='username'
+          component={TextFieldWrapper}
+          label='Username'
+          validate={required}
+          disabled={submitting}
```
Comment (jonambas) [inline src/pages/register/RegisterUserForm.js]: Current UI has this input perma disabled with the users email prefilled

File: src/config/routes.js [hunk 7]
```diff
@@ -81,6 +82,11 @@ const routes = [
     public: true,
     component: SSOPage
   },
+  {
+    path: '/register',
```
Comment (jonambas) [inline src/config/routes.js]: Should we rename this to `/set-password`?

**Pull request merged**